### ISSUE TITLE
Add an ESM edition of `rollup/dist/loadConfigFile`

### ISSUE THREAD

**Issue by mizdra** [rollup/rollup#4146]
### Feature Use Case
I want to use rollup's JavaScript API from Node.js code written in ESM. Currently, the core rollup API (`rollup` module) is provided in both ESM and CJS editions, but only the CJS edition of `rollup/dist/loadConfigFile` module is available. An ESM edition of `rollup/dist/loadConfigFile` module would be useful for users like me.

### Feature Proposal
Add an ESM edition of `rollup/dist/loadConfigFile`.

**Comment by lukastaegert** [comment]
I agree that this would be a very good idea. Unfortunately, there are currently some technical restrictions preventing this:
* `.cjs` configs are loaded via `require`, which of course does not work in an ES module. Once we get rid of all non-ESM capable Node versions, this could be fixed by using a dynamic import here as well.
* bundled configs are loaded via fiddling with `require.extensions`, which definitely does not work in ES modules. There is no simple workaround here as the ESM loader API is still very much experimental and opt-in, and we need this kind of config loading for the new `--configPlugin` API.

We could work around this be putting these things in a dynamically loaded separate CommonJS chunk, but I am not sure this is what you are hoping for.
